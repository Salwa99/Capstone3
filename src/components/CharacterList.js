import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCharactersAction } from '../redux/reducer/reducer';

function CharacterList() {
  const [charNameInput, setCharNameInput] = useState('');
  const dispatch = useDispatch();

  const inputCharNameChangeHandler = (e) => {
    setCharNameInput(e.target.value);
  };

  useEffect(() => {
    dispatch(fetchCharactersAction());
  }, [dispatch]);

  const characters = useSelector((state) => state.characters);
  let filterredCharacters = characters.slice(0, 25);
  if (charNameInput.trim().length > 0) {
    filterredCharacters = characters.filter((item) => (
      item.name.toLowerCase().includes(charNameInput.toLowerCase())
    ));
  }

  return (
    <div className="main">
      <div className="searchField">
        <input type="text" value={charNameInput} placeholder="Enter your favourite Hogwarts character" onChange={inputCharNameChangeHandler} />
      </div>
      <div className="cards">
        <ul>
          {filterredCharacters.map((character) => (
            <li className="mainContainer" key={character.id}>
              <Link to={`/details/${character.name}`}>
                <img src={character.image} alt={character.name} />
                <p>{character.name}</p>
              </Link>
            </li>
          ))}
        </ul>
      </div>

    </div>
  );
}

export default CharacterList;
