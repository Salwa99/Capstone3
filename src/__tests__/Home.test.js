import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CharacterList from '../components/CharacterList';
import store from '../redux/ConfigueStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><CharacterList /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
