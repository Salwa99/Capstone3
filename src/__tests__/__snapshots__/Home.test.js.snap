// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="main"
>
  <div
    className="searchField"
  >
    <input
      onChange={[Function]}
      placeholder="Enter your favourite Hogwarts character"
      type="text"
      value=""
    />
  </div>
  <div
    className="cards"
  >
    <ul />
  </div>
</div>
`;
